torch/_inductor/codegen/triton.py

```diff
@@ -2234,7 +2234,9 @@ def match_mod_div_block(
                 )
                 num_dims = max(
                     2,
-                    len(self.range_tree_nodes),
+                    # range_tree.nodes only includes the entries for the range tree
+                    # len(range_tree.nodes) <= self.range_tree_nodes
+                    len(range_tree.nodes),
```

Comment:
I know this code less well but - would it make sense to try to match for multiple values of `num_dims` @blaine-rister ?

Reply:
@blaine-rister Can I please get a review when you're free? Thanks